File: NzbDrone.Api/ClientSchema/Field.cs
namespace NzbDrone.Api.ClientSchema
{
    public class Field
    {
        public int Order { get; set; }
        public string Name { get; set; }
        public string Label { get; set; }
        public string HelpText { get; set; }
        public string Value { get; set; }
        public string Type { get; set; }
    }
}

File: frontend/src/typings/Field.ts
export interface FieldSelectOption<T> {
  value: T;
  name: string;
  order: number;
  hint?: string;
  parentValue?: T;
  isDisabled?: boolean;
  additionalProperties?: Record<string, unknown>;
}

interface Field {
  order: number;
  name: string;
  label: string;
  value: boolean | number | string | number[];
  section: string;
  hidden: 'hidden' | 'hiddenIfNotSet' | 'visible';
  type: string;
  advanced: boolean;
  privacy: string;
}

export default Field;
